avogadro/qtplugins/labels/labels.cpp

```diff
@@ -0,0 +1,188 @@
+      continue; // Do not process Hydrogens separately
+
+    std::unordered_map<unsigned char, Index>::const_iterator lookUp;
+    lookUp = countMap.find(atomicNumber);
+
+    if (lookUp == countMap.end())
+      countMap[atomicNumber] = 1; // For first occurrence of atom
+
+    else
+      countMap[atomicNumber] += 1; // For other occurrences of atom
+
+    if (!m_showHydrogens)
+      continue;
+
+    Rendering::TextLabel3D* atomLabel = new Rendering::TextLabel3D;
+
+    //  Label non Hydrogen atoms
+    atomLabel->setText(
+      Elements::symbol(atomicNumber) +
+      QString::number(countMap[atomicNumber], 'f', 0).toStdString());
+
+    atomLabel->setRenderPass(Rendering::Overlay3DPass);
+    const Vector3f a1(atom.position3d().cast<float>());
+    const Vector3f& textPos(a1);
+    atomLabel->setAnchor(textPos);
+    geometry->addDrawable(atomLabel);
+
+    Rendering::TextProperties tprop;
+    tprop.setAlign(Rendering::TextProperties::HCenter,
+                   Rendering::TextProperties::VCenter);
+    tprop.setFontFamily(Rendering::TextProperties::SansSerif);
+    tprop.setColorRgb(255, 255, 255);
+    atomLabel->setTextProperties(tprop);
+
+    //  Label Hydrogen atoms
+    const NeighborListType bonds =
+      (const_cast<Molecule&>(molecule))
+        .bonds(atom); // const_cast because bonds() accepts non const objects
+    for (NeighborListType::const_iterator it = bonds.begin(),
+                                          itEnd = bonds.end();
+         it != itEnd; ++it) {
+      const Core::Atom otherAtom =
+        it->atom1().index() != atom.index() ? it->atom1() : it->atom2();
+      if (otherAtom.atomicNumber() == 1) {
+        Rendering::TextLabel3D* hAtomLabel = new Rendering::TextLabel3D;
+        hAtomLabel->setText(
+          "H" + QString::number(hydrogenCount, 'f', 0).toStdString());
+
+        hAtomLabel->setRenderPass(Rendering::Overlay3DPass);
+        const Vector3f a1(otherAtom.position3d().cast<float>());
+        const Vector3f& textPos(a1);
+        hAtomLabel->setAnchor(textPos);
+        geometry->addDrawable(hAtomLabel);
+
+        hAtomLabel->setTextProperties(tprop);
+        hydrogenCount++;
+      }
+    }
+  }
+}
+
+bool Labels::isEnabled() const
+{
+  return m_enabled;
+}
+
+void Labels::setEnabled(bool enable)
+{
+  m_enabled = enable;
+}
+
+QWidget* Labels::setupWidget()
+{
+  if (!m_setupWidget) {
+    m_setupWidget = new QWidget(qobject_cast<QWidget*>(parent()));
+    QVBoxLayout* v = new QVBoxLayout;
+    QCheckBox* check = new QCheckBox(tr("Show multiple bonds?"));
+    check->setChecked(m_multiBonds);
+    connect(check, SIGNAL(clicked(bool)), SLOT(multiBonds(bool)));
+    v->addWidget(check);
+    check = new QCheckBox(tr("Show hydrogens?"));
+    check->setChecked(m_showHydrogens);
+    connect(check, SIGNAL(toggled(bool)), SLOT(showHydrogens(bool)));
+    v->addWidget(check);
+    m_setupWidget->setLayout(v);
+  }
+  return m_setupWidget;
+}
+
+void Labels::multiBonds(bool show)
```

Comment:
This does not seem at all applicable, and I cannot see where it is used in the rest of the class.